Cloud changelog - Sept 28th release (#6644)
mattermost/docs: source/install/cloud-changelog.md

- [Release 2023-08-03](#release-2023-08-03)
- [Release 2023-07-26](#release-2023-07-26)

## Release 2023-09-28

### Improvements

#### User Interface (UI)
 - GitLab plugin v1.7.0 is now available via prepackaged plugins.
 - Added additional reaction options when viewing threads or messages when the sidebar is larger than its minimum width.
 - Added block changes to name, display name, and purpose for direct and group messages.
 - Added a link to notification documentation in the **Notification Settings** modal.
 - Updated the post textbox measurement code to be more reliable.
 - Pre-packaged Calls version v0.19.1.
 - The ``invite`` slash command now supports custom user groups.
 - Re-enabled the remote marketplace functionality (if configured as per ``PluginSettings.EnableRemoteMarketplace``).
 - Added the ability to convert a Group Message channel to a private channel.
 - Group Message channels now behave like Direct Message channels, showing a new mention on every message.

#### Administration
 - Added two new URL parameters to ``GET /api/v4/groups``: ``include_archived`` and ``filter_archived``. Added the ability to restore archived groups from the user groups modal.

>  - Moved the ``request`` package into the public shared folder.

Reviewer: This might go to API changes.

Latest Mattermost Cloud releases:

- [Release 2023-09-28](#release-2023-09-28)
- [Release 2023-09-12](#release-2023-09-12)
- [Release 2023-08-29](#release-2023-08-29)
- [Release 2023-08-09](#release-2023-08-09)
- [Release 2023-08-03](#release-2023-08-03)
- [Release 2023-07-26](#release-2023-07-26)

## Release 2023-09-28

### Improvements

#### User Interface (UI)

>  - GitLab plugin v1.7.0 is now available via prepackaged plugins.

Reviewer 2: Suggestion: Document all prepackaged plugins consistently. Consider the Calls bullet point below as our starting point: ``Pre-packaged Calls version v0.19.1.``

In this case: ``Pre-packaged GitLab plugin version v1.7.0.``

 - Added additional reaction options when viewing threads or messages when the sidebar is larger than its minimum width.
 - Added block changes to name, display name, and purpose for direct and group messages.

>  - Added a link to notification documentation in the **Notification Settings** modal.

Reviewer 2: ```suggestion
 - Added a link to [notification documentation](https://docs.mattermost.com/preferences/manage-your-notifications.html) in the **Notification Settings** modal.
```

 - Updated the post textbox measurement code to be more reliable.
 - Pre-packaged Calls version v0.19.1.
 - The ``invite`` slash command now supports custom user groups.

>  - Re-enabled the remote marketplace functionality (if configured as per ``PluginSettings.EnableRemoteMarketplace``).

Reviewer 2: ```suggestion
 - Re-enabled the remote marketplace functionality, when configured as per ``PluginSettings.EnableRemoteMarketplace`` [documentation](https://docs.mattermost.com/configure/plugins-configuration-settings.html#plugins-enableremotemarketplace).
```

>  - Added the ability to convert a Group Message channel to a private channel.

Reviewer 2: ```suggestion
 - Added the ability to convert a group message channel to a private channel.
```

>  - Group Message channels now behave like Direct Message channels, showing a new mention on every message.

Reviewer 2: ```suggestion
 - Group message channels now behave like direct message channels, showing a new mention for every new message.
```

#### Administration

>  - Added two new URL parameters to ``GET /api/v4/groups``: ``include_archived`` and ``filter_archived``. Added the ability to restore archived groups from the user groups modal.

Reviewer 2: ```suggestion
 - Added 2 new URL parameters to ``GET /api/v4/groups``: ``include_archived`` and ``filter_archived``. Added the ability to restore archived groups from the user groups modal.
```

 - Added new URL Parameter to ``GET /api/v4/groups`` and ``GET /api/v4/groups/:group_id``. ``include_member_ids`` will add all the members ``user_ids`` to the group response objects. You can now also add group members to a channel, any members that are not part of the team can be added to the team through this flow and subsequently added the channel.

### API Changes
 - Added new API endpoint ``GET`` ``/api/v4/channels/<channel-id>/common_teams`` to fetch list of teams common between members of a group message.
 - Added new API endpoint ``POST`` ``/api/v4/channels/<channel-id>/convert_to_channel`` to convert a group message to a private channel.
 - Added a new ``MessageHasBeenDeleted`` hook to the plugin API.
 - Moved the ``request`` package into the public shared folder.

### Bug Fixes
 - Fixed an issue with keyboard support for some menus with submenus.
 - Fixed an issue with a disappearing punctuation when following a group mention.
 - Fixed an issue where compliance export jobs were not able to start after disabling and enabling the compliance export.
 - Fixed a potential read after write issue when loading a license.

### Known Issues

>  - Known issues related to the new feature to convert a Group Message channel to a private channel: [MM-54524](https://mattermost.atlassian.net/browse/MM-54524), [MM-54525](https://mattermost.atlassian.net/browse/MM-54525), [MM-54526](https://mattermost.atlassian.net/browse/MM-54526), [MM-54541](https://mattermost.atlassian.net/browse/MM-54541), [MM-54542](https://mattermost.atlassian.net/browse/MM-54542), [MM-54543](https://mattermost.atlassian.net/browse/MM-54543).

Reviewer 3: Seems https://mattermost.atlassian.net/browse/MM-54524 may not have been an issue to this feature so we can probably remove from here? and https://mattermost.atlassian.net/browse/MM-54543 we've closed as Won't Fix and is a mobile change that won't go out in this Cloud release so we can probably remove as well.

 - ``/invite`` user to a channel incorrectly posts an ephemeral message of ``<no value> added to {channel_name} channel`` [MM-54555](https://mattermost.atlassian.net/browse/MM-54555).

>  - Left-hand side resize option overrides the **Browse/Create Channel** menu [MM-54367](https://mattermost.atlassian.net/browse/MM-54367).

Reviewer 3: This one is only an issue if you have the To-Do plugin installed which I'm not sure you can install in Cloud but we can at least mention that in any case: 
```suggestion
 - Left-hand side resize option overrides the **Browse/Create Channel** menu if To-Do plugin is installed [MM-54367](https://mattermost.atlassian.net/browse/MM-54367).
```
